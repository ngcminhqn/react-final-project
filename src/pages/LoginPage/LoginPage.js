import React from "react";
import {
  ErrorMessage,
  FormStyled,
  ImageStyled,
  LoginContainerStyled,
  TitleScreenStyled,
  TitleStyled,
} from "./LoginPage.styled";
import { IMAGES } from "../../assets";
import { Button, Form, Input } from "antd";
import { useDispatch, useSelector } from "react-redux";
import { authActions } from "../../features/auth/authSlice";
import { useNavigate } from "react-router-dom";
import { PATH_NAME } from "../../constants/pathName";
import Loading from "../../components/Loading/Loading";

const LoginPage = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const { loading, errorMessage } = useSelector((state) => state.auth);

  const onFinish = (values) => {
    const payload = {
      username: values.username,
      password: values.password,
    };
    dispatch(
      authActions.handleLogin({
        data: payload,
        onSuccess: () => {
          navigate(PATH_NAME.HOME, { replace: true });
        },
      })
    );
  };

  return (
    <LoginContainerStyled>
      <Loading isLoading={loading} />
      <TitleStyled>Employee Polls</TitleStyled>
      <ImageStyled src={IMAGES.LoginLogo} alt="logo"></ImageStyled>
      <TitleScreenStyled>Login</TitleScreenStyled>
      <FormStyled
        name="login_employee_form"
        onFinish={onFinish}
        onChange={() => {
          if (errorMessage) {
            dispatch(authActions.resetErrorMessage());
          }
        }}
      >
        <Form.Item
          name="username"
          rules={[{ required: true, message: "Please enter your User!" }]}
          // initialValue={"sarahedo"}
        >
          <Input placeholder="User" />
        </Form.Item>
        <Form.Item
          name="password"
          rules={[{ required: true, message: "Please enter your Password!" }]}
          // initialValue={"password123"}
        >
          <Input type="password" placeholder="Password" />
        </Form.Item>
        {errorMessage ? (
          <Form.Item>
            <ErrorMessage>{errorMessage}</ErrorMessage>
          </Form.Item>
        ) : null}
        <Form.Item>
          <Button type="primary" htmlType="submit">
            Submit
          </Button>
        </Form.Item>
      </FormStyled>
    </LoginContainerStyled>
  );
};

export default LoginPage;
