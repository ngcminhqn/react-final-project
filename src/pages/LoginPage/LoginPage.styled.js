import { Form } from "antd";
import styled from "styled-components";

export const LoginContainerStyled = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100vh;
`;

export const TitleStyled = styled.h1``;

export const TitleScreenStyled = styled.h2``;

export const ImageStyled = styled.img`
  width: 400px;
  height: 400px;
`;

export const ErrorMessage = styled.p``;

export const FormStyled = styled(Form)`
  width: 80%;

  :last-child {
    display: flex;
    justify-content: center;
  }
`;
