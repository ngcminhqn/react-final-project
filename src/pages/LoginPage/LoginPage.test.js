import LoginPage from "./LoginPage";
import { renderWithProviders } from "../../utils/test-utils";

describe("Login", () => {
  it("render login", async () => {
    const { container } = renderWithProviders(<LoginPage />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
