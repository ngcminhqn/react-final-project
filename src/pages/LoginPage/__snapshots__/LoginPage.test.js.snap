// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login render login 1`] = `
<div>
  <div
    class="sc-aYaIB cSpayn"
  >
    <h1
      class="sc-gEvDqW"
    >
      Employee Polls
    </h1>
    <img
      alt="logo"
      class="sc-fqkwJk fjIODc"
      src="login-logo.png"
    />
    <h2
      class="sc-eqUzNf"
    >
      Login
    </h2>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-m4timi sc-iGgVNO hRBvRq"
      id="login_employee_form"
    >
      <div
        class="ant-form-item css-dev-only-do-not-override-m4timi"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-m4timi"
        >
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-m4timi"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input css-dev-only-do-not-override-m4timi ant-input-outlined"
                  id="login_employee_form_username"
                  placeholder="User"
                  type="text"
                  value="sarahedo"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item css-dev-only-do-not-override-m4timi"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-m4timi"
        >
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-m4timi"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input css-dev-only-do-not-override-m4timi ant-input-outlined"
                  id="login_employee_form_password"
                  placeholder="Password"
                  type="password"
                  value="password123"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item css-dev-only-do-not-override-m4timi"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-m4timi"
        >
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-m4timi"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-m4timi ant-btn-primary"
                  type="submit"
                >
                  <span>
                    Submit
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`Login snapshot 1`] = `
<div>
  <div
    class="sc-aYaIB cSpayn"
  >
    <h1
      class="sc-gEvDqW"
    >
      Employee Polls
    </h1>
    <img
      alt="logo"
      class="sc-fqkwJk fjIODc"
      src="login-logo.png"
    />
    <h2
      class="sc-eqUzNf"
    >
      Login
    </h2>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-m4timi sc-iGgVNO hRBvRq"
      id="login_employee_form"
    >
      <div
        class="ant-form-item css-dev-only-do-not-override-m4timi"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-m4timi"
        >
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-m4timi"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input css-dev-only-do-not-override-m4timi ant-input-outlined"
                  id="login_employee_form_username"
                  placeholder="User"
                  type="text"
                  value="sarahedo"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item css-dev-only-do-not-override-m4timi"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-m4timi"
        >
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-m4timi"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input css-dev-only-do-not-override-m4timi ant-input-outlined"
                  id="login_employee_form_password"
                  placeholder="Password"
                  type="password"
                  value="password123"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item css-dev-only-do-not-override-m4timi"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-m4timi"
        >
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-m4timi"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-m4timi ant-btn-primary"
                  type="submit"
                >
                  <span>
                    Submit
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
